### DISCUSSION TITLE
Restrict input directive

### DISCUSSION THREAD

**Discussion by NemirStephan** [Legal-and-General/canopy#300] [Ideas]
I've created a directive to restrict what can be typed into an input and am wondering if this should be a canopy directive. What do people think?

```typescript
import {
  Directive,
  HostListener,
  Input,
  OnChanges,
} from '@angular/core';

/**
 * Restricts input to the specified regEx pattern.
 */
@Directive({
  selector: '[appRestrictInput]',
})
export class RestrictInputDirective implements OnChanges {
  @Input() allowedCharacters: string;

  private regExp: RegExp;

  @HostListener('keypress', ['$event']) onKeyPressed(event) {
    if (!this.regExp.test(event.key)) {
      event.preventDefault();
    }
  }

  ngOnChanges() {
    this.regExp = new RegExp(this.allowedCharacters);
  }
}
```

**Comment by owensgit** [comment]
Looks good, just two questions from me:

- Is it likely this will be used in other apps or does an existing app of ours do something similar already?
- Are there any accessibility concerns? I can't imagine there would be, but it might be worth double checking.

If it's got a chance for reuse, and it's accessible, then I'd be happy to include it 🙂
